Fixes: add Gnome 44 support

jtaala opened the pull request:
This PR fixes reported issues in Gnome 44: fixes #494, fixes #495, closes #490, fixes #523. 

Note: these are compatibility fixes to support Gnome 44 in develop branch (i.e. Gnome 43 should still work fine with these changes).

Gnome 44 has a new (improved?) mechanism for grabs (e.g. grabbing and moving windows), and no longer uses `end_grab_op`.

This PR also adds a polyfill to support Gnome 44's removal of `Meta.later_add` (which is replaced with `global.compositor.get_laters().add(...)`.  

Note: [GJS porting guide](https://gjs.guide/extensions/upgrading/gnome-shell-44.html#meta-later-add-and-meta-later-remove) suggests using `Meta.Laters.add` but in testing the latest Gnome 44 (and what was released in Fedora 38), this was not present.  The implemented approach is used heavily in the latest/official [gnome-shell source code](https://gitlab.gnome.org/search?search=global.compositor.get_laters()&nav_source=navbar&project_id=546&group_id=8&search_code=true&repository_ref=main) (which is why I moved to using that).

~~Note<sup>2</sup>: for the first fix (`end_grab_op`) - it still not quite perfect, it seems to require a second click to activate a window (after moving a window) - but at least it doesn't completely/irreversibly lose mouse click control. 
 Will look at this and see how to improve here.~~  _**fix now implemented for this**_  :heavy_check_mark:

jtaala committed:
Added Polyfil for Meta.add_later.

jtaala committed:
Added optional chain for `.end_grab_op` since in Gnome 44 it's been replaced with new clutter grabs.

jtaala committed:
FIX for another grab-related bug, see
https://github.com/paperwm/PaperWM/commit/88f793c20b6db74e43d98db4699ddcde7a69ea61#r110071632.

jtaala committed:
Add guard/check to showWindow and animateWindow since I ran into a
Window.clone undefined exception during testing.

jtaala commented:
Fiddlesticks... so, I've run into deadends with the grab and drag windows for Gnome 44.  They removed the ability (as far as I can see) to stop a gnome grab operation (which was previously done with `global.display.end_grab_op`).  We need this as we stop the default grab operation and do our own when moving windows.

Gnome did this since now it does it all internally and automatically now... which is simpler... except if you don't want to do grab/drag operation and want to do your own (PaperWM wants to do this).

As mentioned in this PR - it works... but the grab operation is still running after we finish our grab/drag operations... and we then need to "click" again to stop the default gnome drag operation.  It's annoying, but I'm not sure how to (or if we can) fix that in Gnome 44.

Hopefully, we can find another way to end the default grab op before we do our own.  The method/functionality [is still there](https://gitlab.gnome.org/GNOME/mutter/-/blob/44.0/src/compositor/meta-window-drag.c#L1616) but I haven't found a way yet to access it.

If anyone has any ideas here - either how to stop the default gnome `grab op` - or alternatively how to fire off a mouse click after we've done our drag operation let me know.

jtaala committed:
Moved polyfill to kludges.js (where it should be).  Cleaned up and grouped other polyfills.

jtaala committed:
FIX: no longer require second click to "click out" of pointer grab operation to force `end_grab_op`.

jtaala commented:
> If anyone has any ideas here - either how to stop the default gnome `grab op` - or alternatively how to fire off a mouse click after we've done our drag operation let me know.

Huzzah!  so I've implemented a fix for the "needs a second click after pointer grab/moving a window" issue.   Spent a few days trying to regain access to the `end_grab_op` function, then tried rewriting the grab class to fight against the more pervasive gnome 44 grabbing operations.

In the end "if you can't beat em, join em"... we now embrace (lol) the new grab behaviour and then generate a "click out" event automatically to force an internal `end_grab_op` (if needed).

jtaala commented:
To all gnome-44 PaperWM users - could you please test this branch out and report back how it's working for you?  It's working well now in Fedora 38 for me.

```
git fetch --all
git checkout fixes-gnome-44-support
git pull
```
and do a logout/login etc.

jtaala commented:
@Lythenas, if you're on 44, give this a test.  If you're on 43 for your main setup (like me) please test this out (in principle, this PR won't impact/change gnome 43 behaviour as the main changes are polyfills and if `end_grab_op` doesn't exist).

jtaala committed:
Updated readme to add gnome 44 as supported and clarify previous release
support.

jtaala commented:
I updated `README.md` to state gnome 44 support and clarify previous gnome release support.

On a relate note, is this still relevant in `README.md`?  Seems old - but I don't use Ubuntu.  If it's old and not relevant would be happy to remove that section:
https://github.com/paperwm/PaperWM/blob/develop/README.md?plain=1#L37-L58

jtaala commented:
~~uhh - looks like Wayland doesn't like `imports.gi.Atspi.generate_mouse_event`.  Works fine in X11, but hard crashes in Wayland.  Looking for another approach now.~~

Okay, this is now fixed - now using Clutter.virtualInputDevice - which is safe for X11 and Wayland.

jtaala committed:
Revert "FIX: no longer require second click to "click out" of pointer grab operation to force `end_grab_op`."
`gi.Atspi` appears to have severe issues in Wayland - was getting gnome
shell hard crash in wayland with this (X11 is fine).
This reverts commit 566f628c5bf0d985b74b9f685f21554e2f961ebd.

jtaala committed:
FIX (wayland compatible): no longer require second click to "click out" of pointer grab operation to force `end_grab_op`
Now using Clutter virtual device to "click out" - safe to use in X11 and Wayland.

jtaala committed:
Added new GrabOp type introduced in Gnome 44 (MOVING_UNCONSTRAINED)

crispinb commented:
I'll be running the latest commits for the day today. So far so good. Though I wasn't hitting the drag/drop issue and am not clear whether it could be reproduced to test. I don't think I use the mouse enough.

jtaala committed:
FIX: avoid 'click out' for scratch or floating windows (extra click can cause holding onto titlebar and maximizing window).  Moved click out to main loop after activate.

jtaala committed:
FIXES: clickout of grab operation now replicates current cursor position.  Also moved to Meta.LaterType.BEFORE_REDRAW for safety.

Lythenas commented on grab.js:
```diff
@@ -480,6 +480,25 @@ var MoveGrab = class MoveGrab {
+             * until we "click out".  We do this here if needed.
+             */
+            if (!global.display.end_grab_op) {
+                getVirtualPointer().notify_button(Clutter.get_current_event_time(),
+                    Clutter.BUTTON_PRIMARY, Clutter.ButtonState.PRESSED);
+                getVirtualPointer().notify_button(Clutter.get_current_event_time(),
+                    Clutter.BUTTON_PRIMARY, Clutter.ButtonState.RELEASED);
```
Where is this actually clicking? Just the current cursor position? Is there any danger of this clicking on e.g. a button in the UI if we drag the window with Super+Drag?

Like for the scratch window this was causing an issue when dragging from the titlebar. But could this also happen elsewhere?

jtaala replied:
Good question - I believe it's at the current mouse cursor position.  It might be safer to move the cursor to an absolute position.

The original issue can be seen by comments the click mechanism out, and then dragging a window by the titlebar - after dragging is complete, you need to "click again" to end the default gnome grab action (it's interfered with by paperwm's grab/drag system).

jtaala replied:
I'll do some testing here to confirm click position.

jtaala replied:
Hey @Lythenas - good pickup, yes, was clicking somewhere (likely 0,0? or maybe whatever default was... indeterminate?) - in latest commit we now move to ensure cursor is at the current cursor pos before clicking out.

jtaala committed:
FIX: for scratch or floating window click-out causing a re-grab op.  See https://github.com/paperwm/PaperWM/issues/495#issuecomment-1526807903.

paradajz commented:
I'm seeing some very weird issue which is both hard to describe and hard to reproduce. Sometimes when I use my mouse to cycle between opened windows (by clicking on the edge), the WM starts to act as if I started to drag a window. Once I hit this it is very difficult to get out of it - it makes things quite unusable. Last time it happened I just started clicking around and mashing some keys on the keyboard and eventually I managed to get out of it. Very frustrating.

EDIT: Got triggered again simply by pressing Win+Right to switch the active window. Ugh. Think I'll revert to 43 for now.
EDIT2: But I'd be happy to share some logs! Don't know which, though.

jtaala commented:
> Last time it happened I just started clicking around and mashing some keys on the keyboard and eventually I managed to get out of it. Very frustrating.

Yes, I hear you.  Unfortunately, Gnome 44 completely removed the ability to stop using their new grab/drag Clutter grab actions.  This has left us with really no good option.  We're trying to work around this, but it has impacted various things in PaperWM with mouse clicking - I hadn't seen the one with clicking on window previews.

All, unless we can find stable workarounds - I'm starting to get the sinking feeling that the only real option is to fork mutter and re-add the ability to end / disable gnome's default grab/drag actions (which was previously available publicly).  Hopefully we don't need to go there as it's seems crazy just to restore the ability to `end_grab_op` (which is still there but just not publicly available now).  It would also make it more difficult to install PaperWM as we'd need to install a mutter fork to get it to work right...

crispinb commented:
>  I'm starting to get the sinking feeling that the only real option is to fork mutter 

Sinking feeling indeed! This would also complicate matters for anyone already using a patched mutter. Probably a small number - I'm using https://copr.fedorainfracloud.org/coprs/calcastor/gnome-patched/. Though tbh I'd prioritise PaperWM over it.

jtaala commented:
Hey all, I've now implemented some fixes/changes - could you give this one a test? (apologies for lots of test requests - this gnome 44 changes has thrown us for a loop).

Latest changes in this PR seems to be behaving better in my (brief) tests - but others running it for a day (@crispinb, @paradajz  etc) would help confirm.

jtaala commented:
@paradajz, I'm not sure if the issue you're seeing is exactly what we're focusing on atm - sounds like may be related (likely gnome 44's way of detecting resize events to kickoff their new Clutter.grabs?).  We have a resize event that may also need a clickout workaround (?). 

In any case, if you're still on Gnome-44, pull the latest and see if there's been any changes there (I've been trying to reproduce but haven't been able to so far).

jtaala commented:
> @paradajz, I'm not sure if the issue you're seeing is exactly what we're focusing on atm - sounds like may be related (likely gnome 44's way of detecting resize events to kickoff their new Clutter.grabs?). We have a resize event that may also need a clickout workaround (?).

jtaala commented:
Oops - accidentally closed this request (why does github have the close button so close to the comment button).

paradajz commented:
Here's a video of what's going on. This is the latest commit from PaperWM Redux fork ([25d90b46b2d40c84754cbc37b1aedbfea0a80224](https://github.com/PaperWM-redux/PaperWM/commit/25d90b46b2d40c84754cbc37b1aedbfea0a80224), on Wayland. The unintentional drag operation occurs twice in the video.

[Screencast from 2023-04-28 09-08-53.webm](https://user-images.githubusercontent.com/2544094/235079908-c41d299f-c484-4dd2-8c10-1298155af7a9.webm)

jtaala commented:
> The unintentional drag operation occurs twice in the video.

I can't see your mouse in that vid, but I'm assuming this only occurs when clicking the window preview to the left/right edge of screen?  And does it only occur on the furthermost left/edge windows? (i.e. windows against the left/right tiling edge).

It looks like it's tripping a grap op and getting stuck in it.  If you can, can you redo that with `journalctl -f` running in one of those terminals and see if anything weird is getting thrown? (usually is in yellow text if an exception gets thrown).

Thanks @paradajz

paradajz commented:
Ah the mouse wasn't recorded. I've used the GNOME screenshot utility. But I haven't used mouse at all in the video (except when the issue occured) - I've used Super+Left/Right to switch between the windows.

jtaala commented:
> Ah the mouse wasn't recorded. I've used the GNOME screenshot utility. But I haven't used mouse at all in the video (except when the issue occured) - I've used Super+Left/Right to switch between the windows.

It's all good - I can reproduce it with the window previews (clicking lots) but can't reproduce just with Super+Left/Right.  Just want to check, have you disabled the gnome default `Super+Left/Right` shortcuts (which do the left/right tiling?) - like in #516 ?

jtaala commented:
> I can reproduce it with the window previews (clicking lots)

what's weird is that when I reproduce it using the window-preview clicking, it's not from a grab op (e.g. I'm logging on `grab-op-begin` signal) - also nothing weird is thrown in logs.   This should be enough for me to dive into it though.

jtaala commented:
> Here's a video of what's going on. This is the latest commit from PaperWM Redux fork ([25d90b46b2d40c84754cbc37b1aedbfea0a80224](https://github.com/PaperWM-redux/PaperWM/commit/25d90b46b2d40c84754cbc37b1aedbfea0a80224), on Wayland. The unintentional drag operation occurs twice in the video.
> 
>  Screencast.from.2023-04-28.09-08-53.webm

Also, what's weird in that vid is that the naviagator/minimap dissappears and after that the issue occurs (which I'm guessing must be connected to the issue).  @paradajz - any ideas what might be happening there?  I though maybe the trackpad might have been getting tripped (clicked) but I've tried clicking during the fast changes but the minimap doesn't disappear like in that vid.

paradajz commented:
> have you disabled the gnome default `Super+Left/Right` shortcuts (which do the left/right tiling?) - like in #516 ?

Yes, the only default GNOME shortcut I have left is Super+L for screen locking.

> Also, what's weird in that vid is that the naviagator/minimap dissappears and after that the issue occurs (which I'm guessing must be connected to the issue). @paradajz - any ideas what might be happening there? 

Have no idea. However, I just tried to disable the minimap and the issue still occurs.

jtaala commented:
> Here's a video of what's going on. This is the latest commit from PaperWM Redux fork ([25d90b46b2d40c84754cbc37b1aedbfea0a80224](https://github.com/PaperWM-redux/PaperWM/commit/25d90b46b2d40c84754cbc37b1aedbfea0a80224), on Wayland. The unintentional drag operation occurs twice in the video.
> 
>  Screencast.from.2023-04-28.09-08-53.webm



> Here's a video of what's going on. This is the latest commit from PaperWM Redux fork ([25d90b46b2d40c84754cbc37b1aedbfea0a80224](https://github.com/PaperWM-redux/PaperWM/commit/25d90b46b2d40c84754cbc37b1aedbfea0a80224), on Wayland. The unintentional drag operation occurs twice in the video.
> 
>  Screencast.from.2023-04-28.09-08-53.webm

Anyone else at all seeing this?  I tried and tried (doing exactly as @paradajz is doing) and just can't reproduce this.  @paradajz, can you disable any/all other extensions (just for testing) and reproduce (just to rule that out).

crispinb commented:
I haven't been able to get this to happen with a few attempts after reading this thread. And I haven't seen it in my own normal use of PaperWM

smichel17 commented:
I just upgraded to Gnome 44. I encountered the "stuck dragging" issue today (before coming to this thread) on dd3bbb0, which is a few commits behind the latest. Didn't figure out the cause. Tomorrow I'll keep `journalctl -f` running in the background and see what happens.

jtaala commented:
> I just upgraded to Gnome 44. I encountered the "stuck dragging" issue today (before coming to this thread) on [dd3bbb0](https://github.com/paperwm/PaperWM/commit/dd3bbb0b2afadf1adc9ee98b5dee0a26aa7452f2), which is a few commits behind the latest. Didn't figure out the cause. Tomorrow I'll keep `journalctl -f` running in the background and see what happens.

Cool, definitely pull the latest of this PR - that commit you were on had a few "stuck dragging" issues.

smichel17 commented:
Didn't need to wait long.
```
Apr 28 19:08:07 : Can't update stage views actor <clone-container>[<StWidget>:0x5618a44a64d0] is on because it needs an allocation.
Apr 28 19:08:07 : Can't update stage views actor <unnamed>[<ClutterActor>:0x5618a2bdd410] is on because it needs an allocation.
Apr 28 19:08:07 : Can't update stage views actor <unnamed>[<ClutterClone>:0x5618a0f93740] is on because it needs an allocation.
Apr 28 19:08:07 : Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x56189f830b20] is on because it needs an allocation.
Apr 28 19:08:07 : Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f830f00] is on because it needs an allocation.
Apr 28 19:08:34 : Can't update stage views actor <unnamed>[<ClutterClone>:0x56189e1f4e00] is on because it needs an allocation.
Apr 28 19:08:34 : Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x56189f87c3d0] is on because it needs an allocation.
Apr 28 19:08:34 : Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f87eae0] is on because it needs an allocation.
Apr 28 19:08:36 : Can't update stage views actor <clone-container>[<StWidget>:0x5618a44a64d0] is on because it needs an allocation.
Apr 28 19:08:36 : Can't update stage views actor <unnamed>[<ClutterActor>:0x5618a0acacc0] is on because it needs an allocation.
Apr 28 19:08:36 : Can't update stage views actor <unnamed>[<ClutterClone>:0x56189e1f4e00] is on because it needs an allocation.
Apr 28 19:08:36 : Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x56189f87c3d0] is on because it needs an allocation.
Apr 28 19:08:36 : Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f87eae0] is on because it needs an allocation.
Apr 28 19:09:01 : Spurious clutter_actor_allocate called for actor 0x5618a1473bd0/<unnamed>[<ClutterClone>:0x5618a1473bd0] which isn't a descendent of the stage!
Apr 28 19:09:03 : Can't update stage views actor <clone-container>[<StWidget>:0x5618a44a64d0] is on because it needs an allocation.
Apr 28 19:09:03 : Can't update stage views actor <unnamed>[<ClutterActor>:0x5618a0acacc0] is on because it needs an allocation.
Apr 28 19:09:03 : Can't update stage views actor <unnamed>[<ClutterClone>:0x56189e1f4e00] is on because it needs an allocation.
Apr 28 19:09:03 : Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x56189f87c3d0] is on because it needs an allocation.
Apr 28 19:09:03 : Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f87eae0] is on because it needs an allocation.
Apr 28 19:09:06 : Can't update stage views actor <clone-container>[<StWidget>:0x5618a44a64d0] is on because it needs an allocation.
Apr 28 19:09:06 : Can't update stage views actor <unnamed>[<ClutterActor>:0x5618a0acacc0] is on because it needs an allocation.
Apr 28 19:09:06 : Can't update stage views actor <unnamed>[<ClutterClone>:0x56189e1f4e00] is on because it needs an allocation.
Apr 28 19:09:06 : Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x56189f87c3d0] is on because it needs an allocation.
Apr 28 19:09:06 : Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f87eae0] is on because it needs an allocation.
```
It does seem better than before though. This time I was able to just click 3-4 times on the window again and then it un-stuck, where before I had to try too many things to know what broke out of the bugged state.

smichel17 commented:
Two of these seem to appear each time I resize a window:
```
Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f87eae0] is on because it needs an allocation.
```
So I guess likely the line which is relevant to the "stuck" bug is the one in the middle:
```
Spurious clutter_actor_allocate called for actor 0x5618a1473bd0/<unnamed>[<ClutterClone>:0x5618a1473bd0] which isn't a descendent of the stage!
```

jtaala commented:
> Didn't need to wait long.
> 
> ```
> Apr 28 19:08:07 : Can't update stage views actor <clone-container>[<StWidget>:0x5618a44a64d0] is on because it needs an allocation.
> Apr 28 19:08:07 : Can't update stage views actor <unnamed>[<ClutterActor>:0x5618a2bdd410] is on because it needs an allocation.
> Apr 28 19:08:07 : Can't update stage views actor <unnamed>[<ClutterClone>:0x5618a0f93740] is on because it needs an allocation.
> Apr 28 19:08:07 : Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x56189f830b20] is on because it needs an allocation.
> Apr 28 19:08:07 : Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f830f00] is on because it needs an allocation.
> Apr 28 19:08:34 : Can't update stage views actor <unnamed>[<ClutterClone>:0x56189e1f4e00] is on because it needs an allocation.
> Apr 28 19:08:34 : Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x56189f87c3d0] is on because it needs an allocation.
> Apr 28 19:08:34 : Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f87eae0] is on because it needs an allocation.
> Apr 28 19:08:36 : Can't update stage views actor <clone-container>[<StWidget>:0x5618a44a64d0] is on because it needs an allocation.
> Apr 28 19:08:36 : Can't update stage views actor <unnamed>[<ClutterActor>:0x5618a0acacc0] is on because it needs an allocation.
> Apr 28 19:08:36 : Can't update stage views actor <unnamed>[<ClutterClone>:0x56189e1f4e00] is on because it needs an allocation.
> Apr 28 19:08:36 : Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x56189f87c3d0] is on because it needs an allocation.
> Apr 28 19:08:36 : Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f87eae0] is on because it needs an allocation.
> Apr 28 19:09:01 : Spurious clutter_actor_allocate called for actor 0x5618a1473bd0/<unnamed>[<ClutterClone>:0x5618a1473bd0] which isn't a descendent of the stage!
> Apr 28 19:09:03 : Can't update stage views actor <clone-container>[<StWidget>:0x5618a44a64d0] is on because it needs an allocation.
> Apr 28 19:09:03 : Can't update stage views actor <unnamed>[<ClutterActor>:0x5618a0acacc0] is on because it needs an allocation.
> Apr 28 19:09:03 : Can't update stage views actor <unnamed>[<ClutterClone>:0x56189e1f4e00] is on because it needs an allocation.
> Apr 28 19:09:03 : Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x56189f87c3d0] is on because it needs an allocation.
> Apr 28 19:09:03 : Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f87eae0] is on because it needs an allocation.
> Apr 28 19:09:06 : Can't update stage views actor <clone-container>[<StWidget>:0x5618a44a64d0] is on because it needs an allocation.
> Apr 28 19:09:06 : Can't update stage views actor <unnamed>[<ClutterActor>:0x5618a0acacc0] is on because it needs an allocation.
> Apr 28 19:09:06 : Can't update stage views actor <unnamed>[<ClutterClone>:0x56189e1f4e00] is on because it needs an allocation.
> Apr 28 19:09:06 : Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x56189f87c3d0] is on because it needs an allocation.
> Apr 28 19:09:06 : Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f87eae0] is on because it needs an allocation.
> ```
> 
> It does seem better than before though. This time I was able to just click 3-4 times on the window again and then it un-stuck, where before I had to try too many things to know what broke out of the bugged state.

Hmm - sadly there's nothing interesting in the logs there (only allocation exceptions).  Just to confirm - what led up to the behaviour?  Is this the same as @paradajz issue that shows up with just super+left/right (which we've been trying to reproduce?).

jtaala commented:
> ```
> Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x56189f87eae0] is on because it needs an allocation.
> ```
> 
> Two of these seem to appear each time I resize a window. So I guess likely the line which is relevant to the "stuck" bug is the one in the middle
> 
> ```
> Spurious clutter_actor_allocate called for actor 0x5618a1473bd0/<unnamed>[<ClutterClone>:0x5618a1473bd0] which isn't a descendent of the stage!
> ```

Okay, these look interesting.  Is this during a resize operations? (as opposed to a move operation). 

Edit: the last one anyway.  Allocation exceptions also happen on resize events on 43.

smichel17 commented:
> Okay, these look interesting. Is this during a resize operations? (as opposed to a move operation).

Correct.

- No errors normally when switching windows
- When resizing with Meta-R or Meta-F, two `Can't update stage views actor` errors
- Don't know what caused `Spurious clutter_actor_allocate`

---

I just triggered the issue one more time. Both this and the earlier time happened when switching windows with the mouse, so I am not sure if it is the same as @paradajz's issue. I'll take a screen recording next time it happens.

Both last time and this time, somewhere nearby but not at exactly the same time, there was an error from a different extension. I assume it is not related, but I'll share the logs here just on the off chance it is related.

```
Apr 28 19:22:31 : JS ERROR: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.Failed: error occurred in Get
                  _promisify/proto[asyncFunc]/</<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:453:45
                  ### Promise created here: ###
                  getProperty@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/util.js:553:33
                  refreshProperty@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:316:48
                  _refreshOwnProperties/<@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:202:32
                  _refreshOwnProperties@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:200:51
                  _onSignalAsync/refreshPropertiesPromises<@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:256:26
                  _onSignalAsync@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:255:50
                  async*_onSignal@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:214:14
                  _init/<@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/util.js:504:43
Apr 28 19:22:32 : JS ERROR: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.Failed: error occurred in Get
                  _promisify/proto[asyncFunc]/</<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:453:45
                  ### Promise created here: ###
                  getProperty@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/util.js:553:33
                  refreshProperty@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:316:48
                  _refreshOwnProperties/<@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:202:32
                  _refreshOwnProperties@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:200:51
                  _onSignalAsync/refreshPropertiesPromises<@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:256:26
                  _onSignalAsync@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:255:50
                  async*_onSignal@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/appIndicator.js:214:14
                  _init/<@/home/smichel/.local/share/gnome-shell/extensions/appindicatorsupport@rgcjonas.gmail.com/util.js:504:43
```

jtaala commented:
> > Okay, these look interesting. Is this during a resize operations? (as opposed to a move operation).
> 
> Correct.
> 
> * No errors normally when switching windows
> * When resizing with Meta-R or Meta-F, two `Can't update stage views actor` errors
> * Don't know what caused `Spurious clutter_actor_allocate`

Awesome, thanks @smichel17 - haven't really tested the resize op - have been trying to focus on the switching ops.  When you trigged it - it wasn't to do with the window previews (e.g. trying to switch by edge touching)?  I've noticed some things there (and the mouse tiggering a resize grab due to the window border being close... acts differently in 44 - likely due to the grab changes).

smichel17 commented:
I will also mention, I ran this all day during work today and the issue only happened once or twice, vs 3-4 times in just an hour or two of using my personal laptop. Main differences are: 

- Work laptop: wayland, trackball mouse (and probably more keyboard switching than mouse switching)
- Personal laptop: X, trackpad (probably used more to switch windows)

smichel17 commented:
First repro on latest commit was an edge click switch (to window completely offscreen, while preview shows). Second repro was clicking solidly in the middle of a window which was partially on screen, no preview. Second was not a precise click, may have been a tiny drag.

smichel17 commented:
Oh, my. I have 100% repro:

- Open a **flatpak app**. So far tested with Signal, Element, and VSCodium.
- Open some other window. Resize both windows to be larger than 50% of the screen
- Select the non-flatpak window, pushing the flatpak window partially off screen but still visible.
    - MUST be partially off screen. Fully off screen and fully on screen do not 100% reliably trigger the bug.
- Click on the content of the flatpak window (NOT title bar, does not work) ~~and drag a short distance.~~ dragging is not needed, just a click does it.

Clicking in the title bar of an app appears to be the most reliable way to get un-stuck, although it still usually takes a few attempts.

Flatpak explains why I did not encounter the bug much during work. My only flatpak app on that computer is VSCodium; I don't use it for everything; and I'm on a much larger external monitor so "half on screen" doesn't happen as often.

smichel17 commented:
Clicking in the title bar of an app appears to be the most reliable way to get un-stuck, although it still sometimes takes a few attempts.

jtaala commented:
> Oh, my. I have 100% repro:
> 
> * Open a **flatpak app**. So far tested with Signal, Element, and VSCodium.
> * Open some other window. Resize both windows to be larger than 50% of the screen
> * Select the non-flatpak window, pushing the flatpak window partially off screen but still visible.
>   
Nice work!  I think I saw something similar(?) a few times with vscode (the flatpak one) on my test vm - or at least some weird stuff with that.

Will dig into it - I'm soon on a flight from Aus to the US (stuck in airport atm) - so that should give me plenty of time to dig into this one.

tlevani committed:
Remove Gnome tiling left/right shortcuts

smichel17 commented:
Weirdness: My 100% repro is not 100% on my work laptop, even though it is also running Fedora 38 and using VSCodium as a test app. Additionally, I encountered the issue on that computer with the same steps but this time Firefox was the application. So maybe it is not flatpak related, but something that those 3 specific apps are doing… but only some of the time? I'm confused.

jtaala commented:
Haven't had a chance to get back on this - (jet-lag taken more of a toll than I thought... apparently I'm not as young as I used to be..).  Hoping to get a bit more time in the next couple of days.

smichel17 commented:
No worries, you'll get to it when you get to it. I've found that trying to push a side project on myself as a responsibility, rather than working on it when I feel like it, always leads to worse outcomes (stress, burnout) in the long-term.

crispinb commented:
@smichel17 - I can't reproduce with flatpak apps. However I get the stuck-in-drag-mode behaviour under 2 circumstances (sorry not a UI guy so may not have the terminology correct):
- clicking on any window while the workspace stack is active (ie. with a 3 finger trackpad down motion)
- clicking on a full-screen window which is part offscreen (ie. with a normal window adjacent and visible)

I'm a bit unclear on the status of the 'stuck-dragging' windows. An ESC doesn't liberate them, but with a bit of tapping and fiddling eventually I get them back under control and they seem to end up in the scratch layer.

This is a quick fly-by comment. Happy to expand with more text and/or screencaps if the above isn't clear.
@jtaala - to echo @smichel17 's comment above - no worries. There's no SLA here and I'm sure everyone using PaperWM can  only be grateful for your efforts.

cab404 commented:
I also confirm getting frequently and unrecoverably stuck in workspace changing limbo on latest commit in this branch and develop with Gnome 44.1.

tlevani committed:
Add the description for overriding GNOME tiling shortcuts to README

jtaala committed:
Remove Gnome tiling left/right shortcuts (#518)

Based on the discussion in #516 I added gnome 44 shortcut removal using
the same `set-with-backup` bash function used for setting other
recommended gnome-shell settings. I haven't researched if there are any
unwanted consequences of disabling these shortcuts like this, but it
seems to work fine for me. It would be great if someone else with this
issue could also test this before merging. Also, the backup
`restore-gnome-shell-settings-YYYY-MM-DD.sh` file was successfully
created with the original values stored inside.

jtaala commented:
Hey all, thanks for all the feedback.  I'm hoping to get back to working on gnome 44 fixes next week - have noted current issues (still lots with dragging!).  I'm still currently overseas - haven't had much time to work through these but that will change next week.

cab404 commented:
@jtaala I know that might be more hassle than help, but can you tell, where should I start (what's the current suspect) if I want to help with fixing that issue?

jtaala commented:
> @jtaala I know that might be more hassle than help, but can you tell, where should I start (what's the current suspect) if I want to help with fixing that issue?

No probs, it's in `grab.js` - essentially gnome removed the ability to `end_grab_op` (e.g. stop the grab operation so PaperWM can do it's own "tiled" version).

The current workaround is in the `end()` function, where we do a virtual "click" to cause a gnome click-out operation (which causes an internal `end_grab_op`).

It works fairly well, but there are some corner cases (like the `overview` issue that has been noted in this PR comments).

P.S. any help would be appreciated!

jtaala commented:
P.S. I'm cognizant that since gnome 44 has been released (now in Arch too) there's a need to get (at least basic) gnome-44 support up asap!

Given it still might take us a bit to squash these `grab` corner cases (along with testing etc.) - or may even need to rewrite the grab approach (which could take several weeks etc.) - how's this?:

We could disable the manual `grab` operations for now and push this into the main develop branch.  I know this would be quite a big functional regression - e.g. for users who use manual mouse grab / window moving, it would be a hit for sure.  However, that would allow us to get basic gnome-44 support in with everything BUT manual mouse grabbing/moving of windows.

Again, at least things would work and users *shoudn't* get stuck in mid-grab operations (which they can't get out of).

That might allow us a bit of breathing room to come up with a better approach, given the changes in gnome 44.

I could create a new PR called something like `basic-gnome-44-support` or something and we could implement a message when loading saying manual mouse grab ops have been disabled temporarily until we can figure out a workaround.

Thoughts?

Jay.

jtaala commented:
> However, that would allow us to get basic gnome-44 support in with everything BUT manual mouse grabbing/moving of windows.

Just to be more explicit here - moving windows between workspaces via paperWM shortcuts or in overview mode will continue to work as before  - it's basically the moving/readusting of window position within the current workspace (via mouse click-drag) that's currently impacted in gnome 44.

jtaala committed:
FIX: for simple window selection do not initate a move-grab operation
(previous versions did this).  Simplifying this behaviour avoids getting
stuck in a move-grab operations when clicking on a window.

Lythenas commented:
@jtaala I think your suggestion is fine.

We should probably also mention this in the Readme at the top. And maybe also update the list that gnome 43 users should use the v43.1.0 tag (and maybe remove the "experimental").

jtaala commented:
Hey all, call for testing latest commit for gnome-44 (c582a53c71f8b2ee3e21dfff879efc307a4e8b1d).

I'm finally on Gnome 44 for my daily driver and latest commit changed (disabled) a behaviour to avoid one of main sources of getting stuck when selecting a window that's partially off screen.

It's behaving well for me so far - but interested in seeing how it behaves for others and if they still get stuck in grab ops.

randomvariable commented:
Just updated. Mouse grabbing not happening yet. Still can't centre windows without moving the mouse out of focus of the window though.

jtaala commented:
> Still can't centre windows without moving the mouse out of focus of the window though.

Oh... can see an exception there - hadn't noticed that before (I usually don't centre windows but use centre focus mode).  Let me see if can fix that one.

jtaala committed:
FIX: warpPointer method fixed to use Clutter.get_default_backend().
FIX: resizing with decrement/increment size steps now uses
display.focus_window if metaWindow not defined.

jtaala commented:
> Still can't centre windows without moving the mouse out of focus of the window though.

Hey @randomvariable - this *should* now be fixed in latest commit (fixed that and another issue saw when testing that fix) - can you pull/update and test the centering now?

jtaala committed:
Updated `README.md` and pointing gnome 43 to the new gnome-43 branch.
Develop is then targeting gnome-44.

jtaala committed:
Updated PaperWM version number.

jtaala committed:
Small improvement to README.md (formatting).

jtaala commented:
All, I'm actually ready to merge this PR into `develop` branch (the default branch).

I believe I've found the main culprit for getting stuck in drag mode.  I've been thrashing it as much as I can on the latest commit and haven't run into the stuck issues as yet.  Also fixed a few other issues (like centering windows with the center horizontal keybind, and the incrementing/decrementing window widths and height keybinds not working properly - which I think are from other changes in gnome-44).

Given some of these changes/fixes, and the fact that I'm now on gnome 44 (Arch) properly on my main machine, I also updated the `README.md` and created a `gnome-43` branch for `gnome-43` users.  Gnome 43 should work work fine on  `develop` branch, but given some of the changes, and that I can't easily now test it on gnome 43 (since I'm now fully moved to gnome 44) - I thought it was safer to create a branch for gnome 43 and leave it on the commit that worked well (and which is at the `v43.1.0` tag).

In any case, even if there is still a few issues, I'm keen to get this into develop branch, as currently, `develop` branch is just completely broken for **gnome 44**.

jtaala commented:
Please give the latest a good test!

cab404 commented:
It appears to be *mostly* resolved by this!

Although you still end up in the same state — can now get out by clicking empty space on a workspace (clicking windows doesn't work)
![image](https://github.com/paperwm/PaperWM/assets/6453661/7782e64c-0400-4fc6-9b08-c86fb49b1c14)

jtaala commented:
> Although you still end up in the same state — can now get out by clicking empty space on a workspace (clicking windows doesn't work)

Ah, I think I misunderstood you're issue you mentioned, I thought you meant `centre horizontally` keybind (which was broken in gnome 44 so I fixed that).

Just to confirm, you mean when you're in stack mode (like in your pic), we expect that clicking a window should select that window (and get out of stack mode)?

Currently, clicking a window while in stack mode will just ensure it's in the viewport - I can't remember if in gnome 43 the behaviour was to select it and exit stack mode (although I think it was).   Thinking about it, that behaviour (clicking window in stack mode) it does make sense to have it selected and exiting stack mode.  I'll look at (re)implementing this tonight.

letavocado commented:
Hey, @jtaala! 
Switching to the previously active workspace, backward order works as forward order. Though, moving the active window to the previously active workspace in backward order works fine.

Reproducing:
1. Switch to `fixes-gnome-44-support`
2. do a logout/login (maybe twice)

Thank you for Gnome 44 support! :heart:

cab404 commented:
> Just to confirm, you mean when you're in stack mode (like in your pic), we expect that clicking a window should select that window (and get out of stack mode)?

Yup!

>  I'll look at (re)implementing this tonight.

<3!

crispinb commented:
I haven't done any very targeted testing, but in nearly a day of regular use on the 'fixes-gnome-44-support' tip, I've had no problems so far.

randomvariable commented:
> > Still can't centre windows without moving the mouse out of focus of the window though.
> 
> Hey @randomvariable - this _should_ now be fixed in latest commit (fixed that and another issue saw when testing that fix) - can you pull/update and test the centering now?

Yup, it's fixed, thanks! Sorry was out for the rest of yesterday.

jtaala committed:
FIX: in stack mode, if user clicks on window, ensureViewport on that
viewport and exit stack mode.

jtaala commented:
> > Just to confirm, you mean when you're in stack mode (like in your pic), we expect that clicking a window should select that window (and get out of stack mode)?
> 
> Yup!
> 
> > I'll look at (re)implementing this tonight.
> 
> <3!

@cab404, have now fixed this - clicking window while in stack mode now selects that window and exits stack mode.  Feel free to do a `git pull` on this PR branch and  give it a test.

cab404 commented:
> > > Just to confirm, you mean when you're in stack mode (like in your pic), we expect that clicking a window should select that window (and get out of stack mode)?
> > 
> > 
> > Yup!
> > > I'll look at (re)implementing this tonight.
> > 
> > 
> > <3!
> 
> @cab404, have now fixed this - clicking window while in stack mode now selects that window and exists stack mode. Feel free to do a `git pull` on this PR branch and give it a test.

I kinda use a different (and less usable for development) method of rebuilding my nixos system with overridden derivation for PaperWM — how does one use git version of PaperWM normally?)

jtaala commented:
> Hey, @jtaala! Switching to the previously active workspace, backward order works as forward order. Though, moving the active window to the previously active workspace in backward order works fine.
> 
> Reproducing:
> 
> 1. Switch to `fixes-gnome-44-support`
> 2. do a logout/login (maybe twice)
> 
> Thank you for Gnome 44 support! heart

That's weird bananas - wouldn't have expected that to be changed in gnome 44.  How are you switching to the previously active workspace? (paperwm keybind).  I never really used the `Switch to previously active workspace` keybinds - the behaviour I'm seeing looks normal to me(?) - and we haven't changed anything there.  Can you explain the order/behaviour you'd expect?

Might be worth creating a separate issue for this one.

jtaala commented:
> I kinda use a different (and less usable for development) method of rebuilding my nixos system with overridden derivation for PaperWM — how does one use git version of PaperWM normally?)

Oh, I mean just checking out the `fixes-gnome-44-support` branch.  Doing the whole thing from scratch would look like:

```bash
git clone git@github.com:paperwm/PaperWM.git
cd PaperWM
git checkout fixes-gnome-44-support
./install.sh
```
and then logout/login.

Once you're on that branch, you can just do a `git pull` to get any new changes on this branch and then logout/login again.

letavocado commented:
I'm using default keybindings. Now the bug has disappeared, it appears and disappears randomly.

![image](https://github.com/paperwm/PaperWM/assets/31704887/a7184039-e846-4133-a74f-ef59e47c53a7)

jtaala commented:
> I'm using default keybindings. Now the bug has disappeared, it appears and disappears randomly.

Oh poops, okay, please create a new issue for this one and throw those details in it so we can track this, and try to isolate and reproduce.  I haven't seen this, but again, I only really use the swith to above/below workspace keybinds.

PGGB commented:
> have now fixed this - clicking window while in stack mode now selects that window and exists stack mode. Feel free to do a `git pull` on this PR branch and give it a test.

Works great from the testing I've done. Thanks so much for your work!

letavocado commented:
> Oh poops, okay, please create a new issue for this one and throw those details in it so we can track this, and try to isolate and reproduce. I haven't seen this, but again, I only really use the swith to above/below workspace keybinds.

Issue https://github.com/paperwm/PaperWM/issues/523

jtaala committed:
Small change to guard end-grab if clone is null (and showed up in #524,
although that was on gnome 42, which isn't strictly supported in this
branch... but this change won't hurt).

jtaala commented:
Morning all, I'm planning on merging this PR in by end of this week.

Please continue to test this PR as much as possible.  I've been running it solidly and in anger on my main machine for the past 5 days and it's been great (and working without issues).  Hoping others have similar experiences with the latest commit from this branch.

paradajz commented:
I just upgraded to 44 on my main PC and pulled the latest changes from this branch. Ran into dragging issues within seconds.

jtaala commented:
> I just upgraded to 44 on my main PC and pulled the latest changes from this branch. Ran into dragging issues within seconds.

Oh, can you give some more detail on the dragging issues?

Also, just for checking, can you run the `gather-system-info.sh` script and paste output here.

paradajz commented:
> > I just upgraded to 44 on my main PC and pulled the latest changes from this branch. Ran into dragging issues within seconds.
> 
> Oh, can you give some more detail on the dragging issues?
> 
> Also, just for checking, can you run the `gather-system-info.sh` script and paste output here.

Sure, the issue is basically the same as in the video I've posted few days ago. Opened a few windows, switched to one them and got stuck in dragging mode.

```
Please include this information in your bug report on GitHub!
Distribution: openSUSE Tumbleweed
GNOME Shell 44.1
Display server: Xorg
PaperWM branch/tag: fixes-gnome-44-support
PaperWM commit: 477e87a4f04a01ad05245852c0cf645fd19bec0c
Enabled extensions:
- paperwm@hedning:matrix.org
- launch-new-instance@gnome-shell-extensions.gcampax.github.com
- window-list@gnome-shell-extensions.gcampax.github.com
- switcher@landau.fi
- clipboard-history@alexsaveau.dev
- io.github.mreditor.gnome-shell-extensions.scroll-panel
- Vitals@CoreCoding.com
```

jtaala commented:
> > > I just upgraded to 44 on my main PC and pulled the latest changes from this branch. Ran into dragging issues within seconds.
> > 
> > 
> > Oh, can you give some more detail on the dragging issues?
> > Also, just for checking, can you run the `gather-system-info.sh` script and paste output here.
> 
> Sure, the issue is basically the same as in the video I've posted few days ago. Opened a few windows, switched to one them and got stuck in dragging mode.
> 

Dang, I haven't seen any dragging issues in a week now - I've been trying to produce them again.

Anyone else still seeing these dragging issues?

@paradajz I haven't seen the video you mentioned you posted - we'll need to reproduce them to isolate and fix the issues.  Point me to the vid or outline any steps and I'll try reproduce them.

jtaala commented:
> ```
> Please include this information in your bug report on GitHub!
> Distribution: openSUSE Tumbleweed
> GNOME Shell 44.1
> Display server: Xorg
> PaperWM branch/tag: fixes-gnome-44-support
> PaperWM commit: 477e87a4f04a01ad05245852c0cf645fd19bec0c
> Enabled extensions:
> - paperwm@hedning:matrix.org
> - launch-new-instance@gnome-shell-extensions.gcampax.github.com
> - window-list@gnome-shell-extensions.gcampax.github.com
> - switcher@landau.fi
> - clipboard-history@alexsaveau.dev
> - io.github.mreditor.gnome-shell-extensions.scroll-panel
> - Vitals@CoreCoding.com
> ```

Thanks - that looks fine.  Assuming you logged out / logged in again after switching branches etc.

Any logs output would be appreciated too (just output from `journalctl -f /usr/bin/gnome-shell` that covers any "stuck-in-dragging-issue" occurences would do).

paradajz commented:
> > > > I just upgraded to 44 on my main PC and pulled the latest changes from this branch. Ran into dragging issues within seconds.
> > > 
> > > 
> > > Oh, can you give some more detail on the dragging issues?
> > > Also, just for checking, can you run the `gather-system-info.sh` script and paste output here.
> > 
> > 
> > Sure, the issue is basically the same as in the video I've posted few days ago. Opened a few windows, switched to one them and got stuck in dragging mode.
> 
> Dang, I haven't seen any dragging issues in a week now - I've been trying to produce them again.
> 
> Anyone else still seeing these dragging issues?
> 
> @paradajz I haven't seen the video you mentioned you posted - we'll need to reproduce them to isolate and fix the issues. Point me to the vid or outline any steps and I'll try reproduce them.

Here: https://github.com/paperwm/PaperWM/pull/515#issuecomment-1527102511

paradajz commented:
I really don't know how to reproduce. It happens randomly. I just clicked onto a window and the dragging operation started.

paradajz commented:
I've disabled the extension, run the journalctl, enabled PaperWM, switched to the window and the dragging occurred instantly. Nothing special in the log:

```
paradajz@desktop ~
> journalctl -f /usr/bin/gnome-shell
Hint: You are currently not seeing messages from other users and the system.
      Users in the 'systemd-journal' group can see all messages. Pass -q to
      turn off this notice.
svi 18 15:46:59 desktop gnome-shell[8951]: #paperwm disable scratch
svi 18 15:46:59 desktop gnome-shell[8951]: #paperwm disable liveAltTab
svi 18 15:46:59 desktop gnome-shell[8951]: #paperwm disable utils
svi 18 15:46:59 desktop gnome-shell[8951]: #paperwm disable stackoverlay
svi 18 15:46:59 desktop gnome-shell[8951]: #paperwm disable app
svi 18 15:46:59 desktop gnome-shell[8951]: #paperwm disable kludges
svi 18 15:46:59 desktop gnome-shell[8951]: #paperwm disable topbar
svi 18 15:46:59 desktop gnome-shell[8951]: #paperwm disable settings
svi 18 15:46:59 desktop gnome-shell[8951]: #paperwm disable gestures
svi 18 15:46:59 desktop gnome-shell[8951]: #paperwm disable user
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable 1684415573966#
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable tiling
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable navigator
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable keybindings
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable scratch
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable liveAltTab
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable utils
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable stackoverlay
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable app
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable kludges
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable topbar
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable settings
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable gestures
svi 18 15:51:33 desktop gnome-shell[8951]: #paperwm enable user
svi 18 15:51:33 desktop gnome-shell[8951]: Window manager warning: Overwriting existing binding of keysym 2f7259c9 with keysym 2f7259c9 (keycode 31).
svi 18 15:51:33 desktop gnome-shell[8951]: Window manager warning: Overwriting existing binding of keysym ff53 with keysym ff53 (keycode 72).
svi 18 15:51:33 desktop gnome-shell[8951]: Window manager warning: Overwriting existing binding of keysym ff1b with keysym ff1b (keycode 9).
svi 18 15:51:33 desktop gnome-shell[8951]: Window manager warning: Overwriting existing binding of keysym ff51 with keysym ff51 (keycode 71).

```

jtaala commented:
Hmm - yeah, nothing in the logs there.  I just want to rule out branch issues, if you haven't already done so, if you're on the branch, can fully logout/login (or even reboot - gasp) - you're probably aware but disabling/enabling isn't enough after a branch change.

After this, just for testing, I'll get you to fully disable window dragging and see if you're still getting that issue.

paradajz commented:
I did log out, no worries, I actually performed system upgrade, pulled the changes, then restarted my PC.

jtaala commented:
> I did log out, no worries, I actually performed system upgrade, pulled the changes, then restarted my PC.

A few other things to check, 

1. check in `~/.local/share/gnome-shell/extensions` and make sure only one paperwm symlink is there;
2. rerun the `install.sh` script - there's been a few changes to disable a few keybinds that had weird clashes;

if still no luck, can you disable these lines (comment out or delete) in tiling.js:

https://github.com/paperwm/PaperWM/blob/fixes-gnome-44-support/tiling.js#LL1669C1-L1672C67

Then logout/login (or looks like you're on X11 so can do a `alt-f2` and type `r` then press enter).

jtaala commented:
Just want to double-check, if you're still getting drag issues after that last one then there's something else going on.

randomvariable commented:
> Anyone else still seeing these dragging issues?

I've had one issue but no idea how to reproduce it. Got stuck in a state where the scroll wheel would switch between windows horizontally and nothing else. Escape wouldn't get out either, and ultimately had to reboot. Happened once in a whole day.

cab404 commented:
> Anyone else still seeing these dragging issues?

I had experienced them before the clicky drag mode fixes (on 477d546 for sure), but on 477e87a either somehow I don't notice or it is gone

jtaala committed:
On enable, paperwm now disables known keybind clashes, including:

org.gnome.desktop.wm.keybindings switch-group
org.gnome.desktop.wm.keybindings switch-group-backward
org.gnome.desktop.wm.keybindings switch-to-workspace-left
org.gnome.desktop.wm.keybindings switch-to-workspace-right

paradajz commented:
Okay - did a total reinstall - wiped everything and started from scratch - so far so good. Hope it stays that way. :) @jtaala your work is unbelievably appreciated here.

crispinb commented:
I've been f/t on the penultimate commit (477e87a) for the last couple of days. All's been good for me (though I only tend to use the mouse to arrange windows in the scratch layer, so may not come across infrequent dragging problems).

jtaala commented:
> Okay - did a total reinstall - wiped everything and started from scratch - so far so good. Hope it stays that way. :) @jtaala your work is unbelievably appreciated here.

This is interesting, @paradajz, could you give us a bit more info on the reinstall - you didn't have to reinstall the OS did you?  Was it just a reinstall of PaperWM? (and if so how did you do that - delete the extension (just click the `Remove...` button from the `Extensions` app?

What's interesting, is that (I believe) you previously tried this branch on an older commit - and then reverted back to Gnome 43 (since we were still trying to figure out the grabbing issues).  If that's the case, it seems that a checkout of that older commit was still symlinked to `~/.local/share/gnome-shell/extensions`?  I'm just trying to think of possibilities there.

Oh, did you wipe/reset the paperWM gnome settings (in like dconf or something?).

Appreciate any info here so we can record and try to avoid this happening again for others.

jtaala committed:
FIX: found root cause of keybind clashes! Was caused by method signature
mismatch on Gtk.accelerator_parse.

paradajz commented:
To be honest I'm not sure what happened in my setup but I did had two PaperWM versions in extensions directory - the one from this repo and the other one from your redux fork. Right now I'm not sure what was active. I've removed symlinks for both of them and then reinstalled the extension (not the entire OS :)). So far everything working fine - didn't run into dragging issue since yesterday.

sebastianheuer commented:
Happy to confirm that PaperWM works really well again thanks to all the work you put into this branch and also thanks to all the contributions in forms of feedback and bug reports. I am on Fedora 38 and run Gnome 44 on Xorg. No dragging issues, keybindings work reliably, it's a real pleasure to use it (again :D)!

jtaala committed:
Removed Gnome 44 specific settings as not needed now (which disabled keybinds that used to
clash before fixing keybind clash detection).  Also renamed parameter
for keybind fix to reflect Gtk documentation name better.

jtaala committed:
Change to README.md to clarify new features and fixes backporting.

jtaala commented:
Big thanks to everyone for helping with Gnome 44 support in PaperWM.  It was quite a bit more work than first envisioned (those Clutter grab changes threw us for a loop!).

Much appreciated all.

letavocado commented:
Thanks @jtaala! 
You're Awesome! 👓 
PaperWM! :heart:

p.s. :  _`merging soon` can be removed_
